code/game/objects/items/weapons/RSF.dm

```diff
@@ -16,58 +16,75 @@
 							"Paper" = /obj/item/paper,
 							"Pen" = /obj/item/pen,
 							"Dice Pack" = /obj/item/storage/bag/dice,
 							"Cigarette" = /obj/item/clothing/mask/cigarette,
 							"Newdles" = /obj/item/food/chinese/newdles,
 							"Donut" = /obj/item/food/donut,
 							"Chicken Soup" = /obj/item/reagent_containers/drinks/chicken_soup,
-							"Tofu Burger" = /obj/item/food/burger/tofu)
-	var/static/list/rsf_icons = list("Drinking Glass" = image(icon = 'icons/obj/drinks.dmi', icon_state = "glass_empty"),
+							"Tofu Burger" = /obj/item/food/burger/tofu
+							)
+
+	var/static/list/rsf_icons = list(
+							"Drinking Glass" = image(icon = 'icons/obj/drinks.dmi', icon_state = "glass_empty"),
+							"Shot Glass" = image(icon = 'icons/obj/drinks.dmi', icon_state = "shotglass"),
 							"Paper" = image(icon = 'icons/obj/bureaucracy.dmi', icon_state = "paper"),
 							"Pen" = image(icon = 'icons/obj/bureaucracy.dmi', icon_state = "pen"),
 							"Dice Pack" = image(icon = 'icons/obj/dice.dmi', icon_state = "dicebag"),
 							"Cigarette" = image(icon = 'icons/obj/clothing/masks.dmi', icon_state = "cigon"),
 							"Newdles" = image(icon = 'icons/obj/food/food.dmi', icon_state = "chinese3"),
 							"Donut" = image(icon = 'icons/obj/food/bakedgoods.dmi', icon_state = "donut1"),
 							"Chicken Soup" = image(icon = 'icons/obj/drinks.dmi', icon_state = "soupcan"),
-							"Tofu Burger" = image(icon = 'icons/obj/food/burgerbread.dmi', icon_state = "tofuburger"))
-	var/rsf_radial_choice = show_radial_menu(user, src, rsf_icons)
+							"Tofu Burger" = image(icon = 'icons/obj/food/burgerbread.dmi', icon_state = "tofuburger"),
+							"Cigar" = image(icon = 'icons/obj/clothing/masks.dmi', icon_state = "cigaroff"),
+							"Smoked Cheese" = image(icon = 'icons/obj/food/food.dmi', icon_state = "cheesewheel-smoked"),
+							"Edam Cheese" = image(icon = 'icons/obj/food/food.dmi', icon_state = "cheesewheel-edam"),
+							"Blue Cheese" = image(icon = 'icons/obj/food/food.dmi', icon_state = "cheesewheel-blue"),
+							"Camembert Cheese" = image(icon = 'icons/obj/food/food.dmi', icon_state = "cheesewheel-camembert"),
+							"Caviar" = image(icon = 'icons/obj/food/seafood.dmi', icon_state = "caviar")
+							)
+
+	var/static/list/power_costs = list(
+							"Drinking Glass" = POWER_LOW,
+							"Shot Glass" = POWER_LOW,
+							"Paper" = POWER_LOW,
+							"Pen" = POWER_LOW,
+							"Dice Pack" = POWER_LOW,
+							"Cigarette" = POWER_LOW,
+							"Newdles" = POWER_HIGH,
+							"Donut" = POWER_HIGH,
+							"Chicken Soup" = POWER_HIGH,
+							"Tofu Burger" = POWER_HIGH,
+							"Cigar" = POWER_LOW,
+							"Smoked Cheese" = POWER_HIGH,
+							"Edam Cheese" = POWER_HIGH,
+							"Blue Cheese" = POWER_HIGH,
+							"Camembert Cheese" = POWER_HIGH,
+							"Caviar" = POWER_HIGH
+							)
+
+/obj/item/rsf/proc/get_radial_contents()
+	return rsf_icons & rsf_items
+
+/obj/item/rsf/attack_self(mob/user)
+	playsound(loc, 'sound/effects/pop.ogg', 50, 0)
```

Comment:
```suggestion
	playsound(loc, 'sound/effects/pop.ogg', 50, FALSE)
```
the 'vary' var should be a `TRUE/FALSE` tag, rather than 1/0. Code wise its the same, just easier to read